templates/vue/src/components/modals/NodeModal/index.vue

```diff
@@ -690,10 +708,17 @@ export default {
         this.updateNodeCoordinates()
 
         if (this.linkHasThumbnailData) {
-          await this.setLinkData()
+          if (
+            this.node.typeData.thumbnailURL &&
+            (this.type === "add" || this.originalUrl !== this.node.typeData.mediaURL)
+          ) {
+            this.confirmThumbnailsPopup(this.node.typeData.thumbnailURL)
+          } else {
+            await this.setLinkData()
+          }
         }
 
-        if (this.shouldReloadDuration()) {
+        if (this.shouldReloadDuration() && this.kaltura_info === null) {
```

Comment:
Can we move `this.kaltura_info === null` inside of `shouldReloadDuration()` please?